### PR TITLE
Cleanup opencl

### PR DESCRIPTION
**Pull request by t4c1**
## Summary

There is a number of OpenCL kernels and checks that are either unused or can be replaced by kernel generator implementations. This replaces uses and removes them.

## Tests
No changes. Removed stuff was not tested.

## Side Effects
None.

## Release notes
OpenCL: removed unused OpenCL kernels and checks.

## Checklist

- [ ] Math issue #(issue number)

- [ ] Copyright holder: Tadej Ciglarič

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [ ] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [ ] the code is written in idiomatic C++ and changes are documented in the doxygen

- [ ] the new changes are tested

### CODE REVIEW

File: stan/math/opencl/prim/cholesky_decompose.hpp [hunk 1]
```diff
@@ -29,8 +30,11 @@ inline matrix_cl<double> cholesky_decompose(const matrix_cl<double>& A) {
   }
   opencl::cholesky_decompose(res);
   // check_pos_definite on matrix_cl is check_nan + check_diagonal_zeros
-  check_nan("cholesky_decompose (OpenCL)", "A", res);
-  check_diagonal_zeros("cholesky_decompose (OpenCL)", "A", res);
+  check_cl("cholesky_decompose (OpenCL)", "A", res, "not NaN") = !isnan(res);
```
Comment (rok-cesnovar) [inline stan/math/opencl/prim/cholesky_decompose.hpp:33]: Should this check that it is NaN?

Comment (t4c1) [reply stan/math/opencl/prim/cholesky_decompose.hpp:33]: Check is requiring the values not to be nan in order for check to pass. That is in line with what this previously did, right?

Comment (rok-cesnovar) [reply stan/math/opencl/prim/cholesky_decompose.hpp:33]: Yeah. Then I guess the next one should check that the diagonal is not zero.

Comment (t4c1) [reply stan/math/opencl/prim/cholesky_decompose.hpp:33]: However I did the other check backwards. Fixed.

**Review by rok-cesnovar — DISMISSED** [review]
Thanks!

**Review by rok-cesnovar — APPROVED** [review]

**Pull request merged**